File: NotesAndReminders.DataBase/Models/Note.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NotesAndReminders.DataBase.Models
{
    public class Note
    {
        [Key]
        public int NoteId { get; set; }
        public string Header { get; set; } = string.Empty;       //заголовок
        public string Description { get; set; } = string.Empty;  //содержание заметки
        public DateTimeOffset DateCreate { get; set; } = DateTimeOffset.Now; //Дата создания заметка
        [ForeignKey(nameof(NoteId))]
        public ICollection<Tage>? Tags { get; set; } = new List<Tage>();
    }
}

File: notesandreminders.client/src/note/note.ts
import { Reminder } from "../reminder/reminder";
import { Tage } from "../tage/tage";

export class Note {
  constructor(
    public noteId?: number,
    public header?: string,
    public description?: string,
    public dateCreate?: Date,
    public tags?: Tage[],
    public reminderNote?: Reminder ) { }
}
